File: src/main/java/shopping/wishlist/infrastructure/WishlistJpaRepository.java
package shopping.wishlist.infrastructure;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface WishlistJpaRepository extends JpaRepository<WishlistEntity, Long> {

    Page<WishlistEntity> findAllByMemberId(Long memberId, Pageable pageable);

    @Modifying(clearAutomatically = true)
    @Query("UPDATE WishlistEntity w SET w.deletedAt = CURRENT_TIMESTAMP WHERE w.id = :id")
    void deleteById(@Param("id") Long id);

    boolean existsByMemberIdAndProductId(long memberId, long productId);

    boolean existsByMemberIdAndId(Long memberId, Long id);
}

File: src/main/kotlin/shopping/wish/infra/WishListJpaRepository.kt
package shopping.wish.infra

import com.linecorp.kotlinjdsl.support.spring.data.jpa.repository.KotlinJdslJpqlExecutor
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import shopping.wish.domain.WishList

@Repository
interface WishListJpaRepository : JpaRepository<WishList, Long>, KotlinJdslJpqlExecutor
